pkg/provider/pulumi/pulumi.go

```diff
@@ -16,49 +16,115 @@ package pulumi
 
 import (
 	"context"
-	"encoding/json"
 	"errors"
 	"fmt"
-	"time"
+	"strings"
 
-	esc "github.com/pulumi/esc/cmd/esc/cli/client"
+	esc "github.com/pulumi/esc-sdk/sdk/go"
 	corev1 "k8s.io/api/core/v1"
 
 	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
 	"github.com/external-secrets/external-secrets/pkg/utils"
 )
 
 type client struct {
-	escClient    esc.Client
+	escClient    esc.EscClient
+	authCtx      context.Context
 	environment  string
 	organization string
 }
 
 const (
 	errPushSecretsNotSupported       = "pushing secrets is currently not supported by Pulumi"
 	errDeleteSecretsNotSupported     = "deleting secrets is currently not supported by Pulumi"
-	errGettingSecrets                = "error getting secret %s: %w"
-	errUnmarshalSecret               = "unable to unmarshal secret: %w"
 	errUnableToGetValues             = "unable to get value for key %s: %w"
 	errGettingAllSecretsNotSupported = "getting all secrets is currently not supported by Pulumi"
+	errReadEnvironment               = "error reading environment : %w"
+	errPushSecrets                   = "error pushing secret: %w"
+	errInterfaceType                 = "interface{} is not of type map[string]interface{}"
 )
 
 var _ esv1beta1.SecretsClient = &client{}
 
-func (c *client) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
-	x, _, err := c.escClient.OpenEnvironment(ctx, c.organization, c.environment, "", 5*time.Minute)
+func (c *client) GetSecret(_ context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	env, err := c.escClient.OpenEnvironment(c.authCtx, c.organization, c.environment)
 	if err != nil {
 		return nil, err
 	}
-	value, err := c.escClient.GetOpenProperty(ctx, c.organization, c.environment, x, ref.Key)
+	value, _, err := c.escClient.ReadEnvironmentProperty(c.authCtx, c.organization, c.environment, env.GetId(), ref.Key)
 	if err != nil {
 		return nil, err
 	}
-	return utils.GetByteValue(value.ToJSON(false))
+	return utils.GetByteValue(value.GetValue())
 }
 
-func (c *client) PushSecret(_ context.Context, _ *corev1.Secret, _ esv1beta1.PushSecretData) error {
-	return errors.New(errPushSecretsNotSupported)
+func (c *client) PushSecret(_ context.Context, secret *corev1.Secret, data esv1beta1.PushSecretData) error {
+	value := secret.Data[data.GetSecretKey()]
```

Comment:
The secret key is optional. If it's empty we usually push the whole secret. This doesn't seem to be implemented here so I suggest adding a ward if the key is empty say something like error pushing the whole secret is not yet supported.

Reply:
Done!